server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -123,15 +124,19 @@ public Supplier<RetentionLeases> retentionLeasesSupplier() {
      * allocated on both `kind` of nodes.
      */
     public static final Setting<String> INDEX_CODEC_SETTING = new Setting<>("index.codec", "default", s -> {
-        switch (s) {
+        switch (s.toLowerCase(Locale.ROOT)) {
             case "default":
             case "best_compression":
             case "lucene_default":
                 return s;
             default:
-                if (Codec.availableCodecs().contains(s) == false) { // we don't error message the not officially supported ones
+                if (!Codec.availableCodecs().stream().anyMatch(c -> c.equalsIgnoreCase(s))) { // we don't error message the not
+                                                                                              // officially supported ones
                     throw new IllegalArgumentException(
-                        "unknown value for [index.codec] must be one of [default, best_compression] but was: " + s
+                        "unknown value for [index.codec] must be one of [default, best_compression, lucene_default] or "
```

Comment:
If we're allowing for users to set it to lucene_default and/or one of the listed available codecs (Lucene80, Lucene70, etc), this needs to be documented on the static settings for index creation for index.codec setting. 

Ref: https://opensearch.org/docs/2.4/api-reference/index-apis/create-index/#static-index-settings